script/tool/lib/src/format_command.dart

```diff
@@ -206,7 +206,21 @@ class FormatCommand extends PluginCommand {
 
     final String fromPath = relativeTo.path;
 
+    const String handFormattedPragma = '// This file is hand-formatted.';
+
     return files
+        .where((File file) {
```

Comment:
> I made this apply to all files rather than just Dart, mostly because it was easier to plug into the code where `File` objects were still around.

I would much rather this apply only to Dart, as this is strictly worse than `clang-format off` for `clang-format`-formatted files because the latter will prevent people with editors that are configured to auto-format from accidentally formatting the file when editing it. (I really, really wish `dartfmt` would support this for the same reason.)

What's the issue with just adding a `file.basename.endsWith('.dart')` check inside this lambda?

Reply:
it seemed weird to have this code know about file extensions too, but i'm happy to add it if you like